File: HurbChallenge/HurbChallenge/Models/Address.swift
//
//  Address.swift
//  HurbChallenge
//
//  Created by Felipe Alves on 05/07/19.
//  Copyright © 2019 Bolzaniapps. All rights reserved.
//

import Foundation

struct Address: Codable {
    let city: String
    let country: String
    let idAtlasCity: Int?
    let idAtlasCountry: Int?
    let idAtlasNeighborhood: Int?
    let idAtlasState: Int?
    let idCity: Int
    let idCountry: Int
    let idState: Int
    let state: String
    let street: String
    let zipcode: String
    let geoLocation: GeoLocation
    
    enum CodingKeys: String, CodingKey {
        case city
        case country
        case idAtlasCity = "id_atlas_city"
        case idAtlasCountry = "id_atlas_country"
        case idAtlasNeighborhood = "id_atlas_neighborhood"
        case idAtlasState = "id_atlas_state"
        case idCity = "id_city"
        case idCountry = "id_country"
        case idState = "id_state"
        case state
        case street
        case zipcode
        case geoLocation
    }
}

// MARK: -
struct GeoLocation: Codable {
    let lat: Double
    let lon: Double
}

File: app/src/main/java/br/com/rvvaranda/hu/Model/Address.kt
package br.com.rvvaranda.hu.Model


import com.google.gson.annotations.SerializedName

data class Address(
    @SerializedName("address")
    val address: String,
    @SerializedName("city")
    val city: String,
    @SerializedName("country")
    val country: String,
    @SerializedName("countryAlfa2")
    val countryAlfa2: String,
    @SerializedName("country_alfa2")
    val country_Alfa2: String,
    @SerializedName("fullAddress")
    val fullAddress: String,
    @SerializedName("full_address")
    val full_Address: String,
    @SerializedName("geoLocation")
    val geoLocation: GeoLocation,
    @SerializedName("id_atlas_city")
    val idAtlasCity: Any,
    @SerializedName("id_atlas_country")
    val idAtlasCountry: Any,
    @SerializedName("id_atlas_state")
    val idAtlasState: Any,
    @SerializedName("id_city")
    val idCity: Int,
    @SerializedName("id_country")
    val idCountry: Int,
    @SerializedName("id_state")
    val idState: Int,
    @SerializedName("state")
    val state: String,
    @SerializedName("street")
    val street: String,
    @SerializedName("street_name")
    val street_Name: String,
    @SerializedName("streetName")
    val streetName: String,
    @SerializedName("zipcode")
    val zipcode: String


)